src/xr/xr-hand.js

```diff
@@ -0,0 +1,411 @@
+
+        // ray direction
+        var jointL = j1;
+        var jointR = j21;
+
+        if (this._inputSource.handedness === XRHAND_LEFT) {
+            var t = jointL;
+            jointL = jointR;
+            jointR = t;
+        }
+
+        // (A) calculate normal vector between 3 joints: wrist, thumb metacarpal, little phalanx proximal
+        vecA.sub2(jointL._localPosition, this._wrist._localPosition);
+        vecB.sub2(jointR._localPosition, this._wrist._localPosition);
+        vecC.cross(vecA, vecB).normalize();
+
+        // get point between: index phalanx proximal and right phalanx proximal
+        vecA.lerp(j6._localPosition, j16._localPosition, 0.5);
+        // (B) get vector between that point and a wrist
+        vecA.sub(this._wrist._localPosition).normalize();
+
+        // mix normal vector (A) with hand directional vector (B)
+        this._inputSource._rayLocal.direction.lerp(vecC, vecA, 0.5).normalize();
+    }
+
+    // emulate select events by touching thumb tip and index tips
+    if (j4 && j9) {
+        vecA.copy(j4._localPosition);
+        var d = vecA.distance(j9._localPosition);
+
+        if (d < 0.015) { // 15 mm
+            if (! this._inputSource._selecting) {
+                this._inputSource._selecting = true;
+                this._inputSource.fire('selectstart');
+                this._manager.input.fire('selectstart', this._inputSource);
+            }
+        } else {
+            if (this._inputSource._selecting) {
+                this._inputSource._selecting = false;
+
+                this._inputSource.fire('select');
+                this._manager.input.fire('select', this._inputSource);
+
+                this._inputSource.fire('selectend');
+                this._manager.input.fire('selectend', this._inputSource);
+            }
+        }
+    }
+};
+
+/**
+ * @function
+ * @name pc.XrHand#getJointById
+ * @description Returns joint by XRHand id from list in specs: https://immersive-web.github.io/webxr-hand-input/
+ * @param {number} id - id of a joint based on specs ID's in XRHand: https://immersive-web.github.io/webxr-hand-input/
+ * @returns {pc.XrJoint|null} Joint or null if not available
+ */
+XrHand.prototype.getJointById = function (id) {
+    return this._jointsById[id] || null;
+};
+
+Object.defineProperty(XrHand.prototype, 'fingers', {
+    get: function () {
+        return this._fingers;
+    }
+});
+
+Object.defineProperty(XrHand.prototype, 'joints', {
+    get: function () {
+        return this._joints;
+    }
+});
+
+Object.defineProperty(XrHand.prototype, 'tips', {
+    get: function () {
+        return this._tips;
+    }
+});
+
+Object.defineProperty(XrHand.prototype, 'wrist', {
+    get: function () {
+        return this._wrist;
+    }
+});
+
+Object.defineProperty(XrHand.prototype, 'tracking', {
+    get: function () {
+        return this._tracking;
+    }
+});
+
+export { XrHand };
```

Comment:
Should `XrFinger` and `XrJoint` also be exported here?

Reply:
In fact, for the sake of modularity, should `XrFinger` and `XrJoint` be split into separate JS files?

Reply:
Instances of XrFinger and XrJoint are created by the engine.
I can expose then just to allow developers to extend their prototype if they want.

Reply:
Since `XrFinger` and `XrJoint` are exposed in the API reference, we must also export them to the `pc` namespace.